src/models/ComponentStyle.js

```diff
@@ -78,12 +78,14 @@ export default (
      * */
     generateAndInjectStyles(executionContext: Object, styleSheet: StyleSheet) {
       const { isStatic, componentId, lastClassName } = this
-      if (areStylesCacheable && isStatic && lastClassName !== undefined) {
```

Comment:
This is the crux of it, so instead you can replace this: 

`if (areStylesCacheable && isStatic && lastClassName !== undefined) {`

with this:

`if (areStylesCacheable && isStatic && lastClassName !== undefined && styleSheet.hasNameForId(componentId, lastClassName)) {`

a lot simpler, doesn't break our caching and optimisation, and it should have the same effect :+1:

so essentially this is the only change that is needed